lib/services/fanout.go

```diff
@@ -92,16 +98,27 @@ func (f *Fanout) NewWatcher(ctx context.Context, watch types.Watch) (types.Watch
 
 // SetInit sets Fanout into an initialized state, sending OpInit events
 // to any watchers which were added prior to initialization.
-func (f *Fanout) SetInit() {
+// Caller must pass a list of resource kinds confirmed by the upstream event source.
+// As a result of this call, each member watcher will also receive a confirmation
+// based on the provided kinds. Some of the watchers might be closed with an error if resource kinds
+// requested by them weren't confirmed by the upstream event source and they didn't enable
+// partial success mode.
+func (f *Fanout) SetInit(confirmedKinds []types.WatchKind) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 	if f.init {
 		return
 	}
+
+	f.confirmedKinds = make(map[kindSubKind]types.WatchKind)
```

Comment:
```suggestion
	f.confirmedKinds = make(map[kindSubKind]types.WatchKind, len(confirmedKinds))
```

Reply:
Good idea. Fixed in https://github.com/gravitational/teleport/pull/24080/commits/f8b308f1888f6790e3a1554fe7d47fc0d12af81d